### ISSUE TITLE
research: Introduce the option to define conditions on order that require participation of a specific solver or a threshold of solvers that participated in the auction

### ISSUE THREAD

**Issue by cmagan** [cowprotocol/research#22]
### Problem
The Curve cowBurner integration is a prime example.
A protocol integration that is willing to accept zero price guarantees (0 limit price), but would like to get better guarantees that the outcome of the auction reflects a fair market price. Because some of their tokens are pretty niche, in some instances only one solver has been participating in the auction to fill their zero-limit-price orders.

They are one example for users/integrators that wanted to require a minimum threshold of solver participation in an auction.

### Research track
Indicate the research track that this issue relates to. Please also add the relevant track as a label.

**Comment by fleupold** [comment]
My main issue with this feature request is that it makes the optimisation problem solvers have to tackle another degree of magnitude harder. Currently the optimal solution solely depends on information that is available to you at the time of solving.

If an order specified something like: "at least 3 solvers need to bid for my order for me to accept this trade" then the decision for a solver whether or not to include this order depends on what they think other solvers might do (which is a whole new level of guess-game).

Not that it's a no-go, just pointing out this would make solver's lives even harder.

**Comment by fedgiac** [comment]
In a decentralized world there is also a high risk that this can be gamed: people could register their solver, say, three times and appear as if there are three solvers who are submitting solutions, while in reality it's only one.
It's true that they'd have to put up three times the bond, but it would be very hard/a lot of complexity to design the system in a way that can detect this kind of behavior and do the slashing.

**Comment by cmagan** [comment]
In a future of combinatorial auction, we could let solvers ignore these
conditions and apply them after all solutions are collected. Basically
dropping order during auction time.

We can let users define their policies based on recommended lists of
solvers, for example top 2 out of 4 all time volume solver.

**Comment by mfw78** [comment]
Considering the case of _requiring_ a specific solver - this seems very much to basically just being the same as an OTC with a potential option for another solver doing better. IMO it seems that this would be more suited for the user to _not_ use the batch auction mechanism and just engage the solver in an OTC deal.

> In a future of combinatorial auction, we could let solvers ignore these conditions and apply them after all solutions are collected. Basically dropping order during auction time.

Having solvers supporting nuances seems a difficult ask, given the current inertia displayed in having integrations for features such as CoW AMM. For these kinds of designs, it would be possible to implement this feature as a programmatic order (in combination with solver buy-in, the consuming protocol running some API of sorts, and a robust method for which CoW Protocol is able to signal which addresses are eligible for surplus - i.e. similar to CoW AMM).

Then in which case:
1. Integrating solvers poll the protocol's API to commit to submitting a solution.
2. Towards the end of the batch auction, solvers that have committed, request for a signature from the API.
3. If (2) has been satisfied, include the respective JIT order in the solution.

In the above case, this would allow for novel solutions to be built based on arbitrary conditions, and keep this kind of logic very separate from the core protocol, but maintain a high level of extensibility.

Cons: consuming protocol needs to run an API (arguably should be the case with a watch-tower already of sorts, kind of overhead), and solvers need specific integration (would be required nonetheless with previously proposed solutions).

**Comment by cmagan** [comment]
> Considering the case of requiring a specific solver - this seems very much to basically just being the same as an OTC with a potential option for another solver doing better. IMO it seems that this would be more suited for the user to not use the batch auction mechanism and just engage the solver in an OTC deal.

I don't agree with this statement. 
I think that protocols want to get matched in the auction, but just want to make sure there's enough participation from solvers. We can offer lists of solvers (say top 10 solvers), and the user can define at least 3 out of 10 participating. 


> Having solvers supporting nuances seems a difficult ask

I agree! 
What about applying these conditions at auction run time so solvers are not required to consider them when proposing solutions?

**Comment by fleupold** [comment]
I agree that in a combinatorial auction world solvers could just solve the order as if it was valid (and still report alternative solutions without it). The option of saying "m out of n fixed solvers" need to participate seems to be addressing the sybil attack concern.
